src/utils/govPayMetadata.ts

```diff
@@ -44,14 +47,14 @@ const parseMetadata = ({
 }): FormattedMetadata => {
   let entries: [string, GovPayMetadataValue][] = [];
 
-  entries = metadata.map(({ key, value }) => {
+  entries = metadata.map(({ key, value, type }) => {
     // ITP data is set at runtime by user journey, and not read from passport directly
     if (key === ITP_KEY) return [ITP_KEY, paidViaInviteToPay];
 
-    if (!isPassportValue(value)) return [key, value];
+    if (!isPassportValue(value, type)) return [key, value];
 
-    // Remove "@" prefix
-    const passportKey = (value as string).substring(1);
+    // Remove "@" prefix (if present)
+    const passportKey = value.startsWith("@") ? value.substring(1) : value;
```

Comment:
Now that we have a "static" and "data" distinction, is there ever a valid reason you'd want a "static" value that starts with "@" to be submitted as-is without stripping? Eg should this be a scenario in the `mockMetadata` ?